src/main/java/com/limechain/rpc/server/RpcApp.java

```diff
@@ -21,7 +21,8 @@
     "com.limechain.rpc.config",
     "com.limechain.rpc.methods",
     "com.limechain.rpc.server",
-    "com.limechain.storage"
+    "com.limechain.storage",
+    "com.limechain.runtime.builder",
```

Comment:
This is a shortcut (doesn't necessarily scale well). I prioritized the explicit intent that `RuntimeBuilder` is not a class that is supposed be independently used, since it only makes sense as a spring bean... that's its whole purpose.

In any case, that's a collision of design principles - in the long term I'd EITHER get rid of this explicit component scan and migrate all actual beans into proper bean classes; OR stick to the current  paradigm and move the logic currently residing in `RuntimeBuilder` into wherever otherwise appropriate.

Reply:
This also has to do with the existence of the `com.limechain.runtime.builder` package. I really don't like it as a subpackage of `com.limechain.runtime` as this `RuntimeBuilder` generally should live outside of it. I couldn't find a proper super-package to put it into, so that's why I sub-packaged.

Reply:
In the case of `RpcApp` is there a reason why we have it under `com.limechain.rpc.server`? If we bring it a level higher  into the "root directory" of the rpc we can skip adding `com.limechain.rpc.XXX`. 
Regarding the other beans, if we do not have the `RuntimeBuilder` as a RPC specific class (under the rpc packages) I do not see an issue of scanning it's package separately. When we point to a specific package containing only a single file that does not bring much overhead.

Reply:
My issue is not with overhead; I'm rather wary of the potential packaging chaos this would introduce. It's not intuitive and doesn't really follow any kind of structural convention / principle.

Regarding your first question - I'm not sure there's any concrete design intention behind that or how up-to-date that is with the current state of the source base. I feel more like that's how it once started and it's just stuck around till now.

Anyway, I just wanted to point this out as a potential "not nice thing" that's currently happening. It will become more apparent if you think about all the places that rely on `AppBean.getBean` (meaning that those classes are essentially spring components), that are sprinkled around everywhere. This `RuntimeBuilder` is just one of them that I've made explicit in this PR.

Reply:
My comment was not solely for mentioning overhead. See [this](https://docs.spring.io/spring-boot/docs/1.5.14.RELEASE/reference/html/using-boot-structuring-your-code.html#using-boot-locating-the-main-class). Moving the main class would be a step to improve the "structure chaos".
In this specific case I do not see a problem with the builder. Each bean has it's own domain (to some extent), thus lives in a different package. If in the future we need to create more "builders" we can easily have a factory for builders and have that in a separate package. In that way we'd always go through a single class when requesting a specific component (builder).

Reply:
I completely agree with you. I think you're saying what I implicitly meant with my EITHER clause above, so we're more or less on the same page 😄 The thing is: this entails some non-trivial refactoring as well, which I deemed out of scope. My main issue was that we've got it mixed right now - a lot of actual beans are not explicit beans, but serve (and can only be used) as such - those are the classes I meant above, which use `AppBean.getBean`. And I suspect no "structuring"/"packaging" has been thought of explicitly whilst introducing those `AppBea.getBean` usages, so if we go the way of explicitly listing all beans in the component scan annotation (if we suppose that we undertake the refactoring of making beans explicit), we're going to end up with a rather messy incoherent list. As an example, what I've introduced here with this `runtime.builder` **sub**package doesn't make a lot of sense. 

Otherwise, I'm entirely on the same page about each bean having its own domain. It's just that those domains are not well outlined right now.

Regarding the moving of `RpcApp` class, my response here:
> Regarding your first question - I'm not sure there's any concrete design intention behind that or how up-to-date that is with the current state of the source base. I feel more like that's how it once started and it's just stuck around till now.

essentially boils down to "I'm not even sure and haven't looked into it deeply enough".